tests-cuda/test_1809-cuda-overload.py

```diff
@@ -0,0 +1,38 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import numpy as np  # noqa: F401
+import pytest  # noqa: F401
+
+import awkward as ak  # noqa: F401
+
+numba = pytest.importorskip("numba")
+
+from numba.extending import overload_method
+from numba import cuda, types
+
+from numba import config
+config.CUDA_LOW_OCCUPANCY_WARNINGS = False
+config.CUDA_WARN_ON_IMPLICIT_COPY = False
+
+@overload_method(types.Array, 'sum', target='cuda')
+def array_sum(arr):
+    if arr.ndim != 1:
+        # Only implement 1D for this quick example
+        return None
+
+    def sum_impl(arr):
+        res = 0 
+        for i in range(len(arr)):
+            res += arr[i]
+        return res 
+    return sum_impl
+
+@cuda.jit
+def f(arr):
+    print("Sum is", arr.sum())
```

Comment:
@gmarkall - I'm having a problem with overloading this method copied from your notebook. Could you, please, check if I'm missing something? Thanks.

Here is the error:
```python
>               raise newerr.with_traceback(tb)
E               numba.core.errors.LoweringError: Failed in cuda mode pipeline (step: native lowering)
E               No definition for lowering (<class 'numba.core.types.npytypes.Array'>, 'sum')(array(int64, 1d, C),) -> int64
E               
E               File "tests-cuda/test_1809-cuda-overload.py", line 32:
E               def f(arr):
E                   print("Sum is", arr.sum())
E                   ^
E               
E               During: lowering "$10call_method.4 = call $8load_method.3(func=$8load_method.3, args=[], kws=(), vararg=None, varkwarg=None, target=None)" at /eos/home-y/yana/Projects/PR1809/awkward/tests-cuda/test_1809-cuda-overload.py (32)
```

Reply:
I'm getting the same failure with Numba 0.56.3 - I think that correct operation of the overload API requires a fix that only went into 0.56.4: https://github.com/numba/numba/pull/8570.

Unfortunately Numba 0.56.4 is not yet available in conda-forge (see https://github.com/conda-forge/numba-feedstock/pull/108) or the default channels, but it can be installed from the Numba channel:

```
$ mamba search -c numba numba

# ... lots of output snipped ...

numba                         0.56.4 np1.11py3.7hc13618b_g288a38bbd_0  numba               
numba                         0.56.4 np1.11py3.8hc13618b_g288a38bbd_0  numba               
numba                         0.56.4 np1.16py3.9h59fbc97_g288a38bbd_0  numba               
numba                         0.56.4 np1.21py3.10ha54aa7d_g288a38bbd_0  numba     
```

Reply:
Thanks! I confirm that it works - after pip installing `numba` 0.56.4.